BangazonAPI/BangazonAPI/Controllers/PaymentTypeController.cs

```diff
@@ -0,0 +1,232 @@
+                    {
+                        paymentType = new PaymentType
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                            AcctNumber = reader.GetInt32(reader.GetOrdinal("AcctNumber")),
+                            Name = reader.GetString(reader.GetOrdinal("Name")),
+                            CustomerId = reader.GetInt32(reader.GetOrdinal("CustomerId")),
+                            IsArchived = reader.GetBoolean(reader.GetOrdinal("IsArchived"))
+                        };
+                        paymentTypes.Add(paymentType);
+                    }
+                    reader.Close();
+                    return Ok(paymentTypes);
+                }
+            }
+        }
+
+        //GET Code for Single Payment Type (Is[NOT]Archived?):
+        [HttpGet("{id}", Name = "PaymentType")]
+        public async Task<IActionResult> GetSinglePaymentType([FromRoute] int id)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = $"SELECT Id, AcctNumber, Name, CustomerId, IsArchived from PaymentType WHERE Id=@id";
+                    cmd.Parameters.Add(new SqlParameter("@id", id));
+                    SqlDataReader reader = cmd.ExecuteReader();
+                    PaymentType paymentTypeToDisplay = null;
+                    while (reader.Read())
+                    {
+                        {
+                            paymentTypeToDisplay = new PaymentType
+                            {
+                                Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                                AcctNumber = reader.GetInt32(reader.GetOrdinal("AcctNumber")),
+                                Name = reader.GetString(reader.GetOrdinal("Name")),
+                                CustomerId = reader.GetInt32(reader.GetOrdinal("CustomerId")),
+                                IsArchived = reader.GetBoolean(reader.GetOrdinal("IsArchived"))
+                            };
+                        };
+                    };
+                    reader.Close();
+                    return Ok(paymentTypeToDisplay);
+                }
+            }
+        }
+
+        //  POST: Code to CREATE a Payment Type:
+        [HttpPost]
+        public async Task<IActionResult> PostPaymentType([FromBody] PaymentType paymentType)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = $@"INSERT INTO PaymentType (AcctNumber, Name, CustomerId, IsArchived)
+                                                    OUTPUT INSERTED.Id
+                                                    VALUES (@AcctNumber, @Name, @CustomerId, 1)";
```

Comment:
So this is exactly the right idea, except that the bit of 1 converts to `true` (`false` is 0), which means that whenever we're creating a new payment type it automatically sets `isArchived` to `true`. We want it to start out as false and change to true when they delete it.

Reply:
Thanks!  Made applicable changes and updated to [current version of] master!